anomalib/utils/callbacks/visualizer_callback.py

```diff
@@ -87,41 +85,21 @@ def _add_to_logger(
                 if isinstance(available_loggers[log_to], ImageLoggerBase):
                     logger: ImageLoggerBase = cast(ImageLoggerBase, available_loggers[log_to])  # placate mypy
                     logger.add_image(
-                        image=visualizer.figure,
+                        image=image,
                         name=filename.parent.name + "_" + filename.name,
                         global_step=module.global_step,
                     )
 
-    def generate_visualizer(self, outputs) -> Iterator[Visualizer]:
-        """Yields a visualizer object for each of the images in the output."""
-        for i in range(outputs["image"].size(0)):
-            visualizer = Visualizer()
-
-            image = Denormalize()(outputs["image"][i].cpu())
-            anomaly_map = outputs["anomaly_maps"][i].cpu().numpy()
-            heat_map = superimpose_anomaly_map(anomaly_map, image, normalize=not self.inputs_are_normalized)
-            pred_score = outputs["pred_scores"][i].cpu().numpy()
-            pred_label = outputs["pred_labels"][i].cpu().numpy()
-
-            if self.task == "segmentation":
-                pred_mask = outputs["pred_masks"][i].squeeze().int().cpu().numpy() * 255
-                vis_img = mark_boundaries(image, pred_mask, color=(1, 0, 0), mode="thick")
-                visualizer.add_image(image=image, title="Image")
-                if "mask" in outputs:
-                    true_mask = outputs["mask"][i].cpu().numpy() * 255
-                    visualizer.add_image(image=true_mask, color_map="gray", title="Ground Truth")
-                visualizer.add_image(image=heat_map, title="Predicted Heat Map")
-                visualizer.add_image(image=pred_mask, color_map="gray", title="Predicted Mask")
-                visualizer.add_image(image=vis_img, title="Segmentation Result")
-            elif self.task == "classification":
-                visualizer.add_image(image, title="Image")
-                if pred_label:
-                    image_classified = add_anomalous_label(heat_map, pred_score)
-                else:
-                    image_classified = add_normal_label(heat_map, 1 - pred_score)
-                visualizer.add_image(image=image_classified, title="Prediction")
-
-            yield visualizer
+    @staticmethod
+    def _show_image(title, image):
+        cv2.imshow(title, image)
+        cv2.waitKey(0)
+        cv2.destroyAllWindows()
+
+    @staticmethod
+    def _save_image(file_path, image):
+        file_path.parent.mkdir(parents=True, exist_ok=True)
+        cv2.imwrite(str(file_path), image)
```

Comment:
I feel like it would be better if `show_image` and `save_image` belong to `Visualizer` class

Reply:
The reason being that `Visualizer` already have `add_image`. It would be more logical to me if `show_image` is part of `Visualizer` object as well.